src/test/java/com/connexta/search/common/SearchTermITests.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.common;
+
+import static com.connexta.search.common.configs.SolrConfiguration.*;
+import static com.connexta.search.common.configs.SolrConfiguration.SOLR_COLLECTION;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import com.connexta.search.index.Index;
+import com.connexta.search.index.IndexManager;
+import com.connexta.search.query.QueryManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+/**
+ * The purposes of this test class are to verify that all search terms can be added to index without
+ * causing an exception and that all search terms can be a part of a valid query sent to the index.
+ */
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@Testcontainers
+@DirtiesContext
+public class SearchTermITests {
+
+  private static final int SOLR_PORT = 8983;
+
+  @Container
+  public static final GenericContainer solrContainer =
```

Comment:
✏️ We can optionally use an embedded solr server.

Reply:
I think we should hold off on that for now. We have a ticket to add that improvement in our backlog.

Reply:
What I'm about to write is off-topic. Also, this isn't the right the place for the discussion, but putting our integration tests inside of the code base itself has always given me heart burn -- even in early DDF days. I like unit tests to accompany the source code, but I wish it stopped there. I would rather spin up containers, services, or servers and test them with an external tool. </rant>

Reply:
I do not think it is bad to have integration tests live next to our code. Integration tests in the new sense is not setting up a full instance like DDF integration tests. Plus we just have to make sure ours are not slow and flaky unlike DDF!

Reply:
@peterhuffer I think the world agrees with you. Otherwise things like Pax Exam and Spring Boot Test  would not exist, right? 
My contrarian opinion comes from having been tripped up on things like subtle differences between RestTemplate and TestRestTemplate. In another case, we also spend several days figuring why an itest would not work. Eventually, Josh stuck a `@DirtiesContext` annotation on the test and everything worked perfectly. 
I feel like Pax Exam and Spring Boot Test lure you in with the promise of easier integration tests, but the little gotchas end up costing you more time than you expected. Investing in a purpose-built tool has its own drawbacks, but it is common thing to do in the industry.